core/src/main/java/org/apache/druid/java/util/http/client/pool/ResourcePool.java

```diff
@@ -119,11 +119,17 @@ public V get()
         return value;
       }
 
+      @Override
+      public boolean isPresent()
+      {
+        return !returned.get();
+      }
+
       @Override
       public void returnResource()
       {
         if (returned.getAndSet(true)) {
-          log.warn(StringUtils.format("Resource at key[%s] was returned multiple times?", key));
+          log.warn("Resource at key[%s] was returned multiple times?", key);
```

Comment:
I am little unclear on this. Isn't this log here to indicate that a resource which is taken once should only be returned once? If we always need to check `isPresent()` before calling `returnResource`, this warn log probably doesn't serve much purpose.

Reply:
I was thinking the log message is useful in situations where the caller thinks they are returning a resource for the only time. Most calls to returnResource are still not guarded by isPresent, for this reason.

However for exceptionCaught specifically, it may be called multiple times on the same response. So we need some way to account for that. I thought about removing the log message completely, but then chose to keep it and use isPresent, since I figured it was still useful in those other cases.